lib/cocoapods/installer/analyzer.rb

```diff
@@ -496,8 +514,6 @@ def generate_pod_targets(specs_by_target)
             test_dependencies = transitive_dependencies_for_specs(target.specs.select(&:test_specification?), target.platform, all_specs).group_by(&:root)
             target.dependent_targets = filter_dependencies(dependencies, pod_targets_by_name, target)
             target.test_dependent_targets = filter_dependencies(test_dependencies, pod_targets_by_name, target)
-            # Test dependendent targets include our own target as a test dependency.
```

Comment:
this is now manually handled instead because this causing confusion on whether a pod is a test only pod or not.